lexnlp/utils/tests/test_text_vectors.py

```
@@ -111,3 +111,85 @@ def test_stop_past_end_is_safe(self):
+    """Tests that exercise the PR-specific _as_string_array behaviour."""
+
+    def test_ndarray_input_uses_string_dtype(self):
+        """When given an ndarray, _as_string_array must return an array with
+        StringDType (not the legacy fixed-width np.str_)."""
+        from lexnlp.utils.text_vectors import _as_string_array
+
+        inp = np.array(["hello", "world"])
+        out = _as_string_array(inp)
+        self.assertIsInstance(out, np.ndarray)
+        # StringDType should compare equal to an instance of StringDType
+        self.assertIsInstance(out.dtype, np.dtypes.StringDType)
```

Pylint (4.0.5)

[error] Module 'numpy.dtypes' has no 'StringDType' member

(E1101)